functests/3_performance_status/status.go

```diff
@@ -0,0 +1,74 @@
+package __performance_status
+
+import (
+	"context"
+	"fmt"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	testutils "github.com/openshift-kni/performance-addon-operators/functests/utils"
+	testclient "github.com/openshift-kni/performance-addon-operators/functests/utils/client"
+	"github.com/openshift-kni/performance-addon-operators/functests/utils/mcps"
+	"github.com/openshift-kni/performance-addon-operators/functests/utils/nodes"
+	"github.com/openshift-kni/performance-addon-operators/functests/utils/profiles"
+	v1 "github.com/openshift/custom-resource-status/conditions/v1"
+	machineconfigv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var _ = Describe("Status testing of performance profile", func() {
+	var workerRTNodes []corev1.Node
+	var err error
+
+	BeforeEach(func() {
+		workerRTNodes, err = nodes.GetByRole(testutils.RoleWorkerCNF)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(workerRTNodes).ToNot(BeEmpty())
+	})
+
+	Context("[rfe_id:28881][performance] Performance Addons detailed status", func() {
+		var nodeAnnotationConfig = "machineconfiguration.openshift.io/currentConfig"
+		var currentConfig string
+		nodeLabel := map[string]string{fmt.Sprintf("%s/%s", testutils.LabelRole, testutils.RoleWorkerCNF): ""}
+		annotations := workerRTNodes[0].GetAnnotations()
+		for k, v := range annotations {
+			if k == nodeAnnotationConfig {
+				currentConfig = v
+			}
+		}
+
+		It("[test_id:29673] Machine config pools status tied to Performance Profile", func() {
+
+			currentConfigAnnotation := map[string]string{
+				nodeAnnotationConfig: currentConfig,
+			}
+			updateAnnotation := map[string]string{
+				nodeAnnotationConfig: "",
+			}
+
+			// Empty the value of "machineconfiguration.openshift.io/currentConfig" for node with worker-cnf label
+			By("Applying changes in machineconfiguration currentConfig of CNF worker node")
+			Expect(testclient.Client.Patch(context.TODO(), &workerRTNodes[0],
+				client.ConstantPatch(
+					types.JSONPatchType,
+					[]byte(fmt.Sprintf(`[{ "op": "replace", "path": "/metadata/annotations/%s", "value": %s }]`, nodeAnnotationConfig, updateAnnotation)),
+				),
+			)).ToNot(HaveOccurred())
+			// Wait until worker-cnf MCP is in degraded state and get condition reason
+			mcps.WaitForCondition(testutils.RoleWorkerCNF, machineconfigv1.MachineConfigPoolDegraded, corev1.ConditionTrue)
+			profileMessage := profiles.GetProfileMessage(nodeLabel, v1.ConditionDegraded)
+			// Verify the status reason of performance profile
+			conditionReason := "1 nodes are reporting degraded status on sync"
```

Comment:
I think we had a misunderstand here from my last review:
I think we should avoid looking for hardcoded strings we are not responsible for. When MCP changes this string, this test will break, for no good reason.
We should get this string from the MCP condition, and see if we use the same. Makes sense?

Reply:
Yes, I misunderstood. Will fix it.

Reply:
Done.